solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -1831,21 +1814,27 @@ private void fetch() throws Exception {
     private int fetchPackets(FastInputStream fis) throws Exception {
       byte[] intbytes = new byte[4];
       byte[] longbytes = new byte[8];
+      boolean isContentReceived = false;
       try {
         while (true) {
+          if (fis.peek() == -1) {
+            return NO_CONTENT;
```

Comment:
Shouldn't we add this log here as seenbelow:             if (!isContentReceived) log.warn("No content received for file: {}", fileName);